cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperator.java

```diff
@@ -3456,4 +3473,37 @@ private String getStringHash(String toBeHashed)  {
             throw new RuntimeException("Failed to create SHA-512 MessageDigest instance");
         }
     }
+
+    public static class BrokerConfigChange extends RestartReason {
```

Comment:
Is it `public` for a reason? We should probablly document at least why a subclass is necessary. And I think it might be cleaner froma layering PoV if:

* `RestartReason` was made topic level
* `BrokerConfigChange` was moved to the same package as `KSO`.